contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md

```diff
@@ -0,0 +1,280 @@
+   the scheduler cache (`AssumePod` operation) and run plugins from the `Reserve`
+   and `Permit` extension point in that order. In case either of the plugins fails,
+   end the current scheduling cycle, increase relevant metrics and handle
+   the scheduling error through the `Error` handler.
+1. Upon successfully running all extension points, proceed to the binding cycle.
+   At the same time start processing another pod (if there’s any).
+
+### Binding cycle
+
+Consists of the following four extension points ran in the same order:
+- [WaitOnPermit](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L560)
+  (internal API): Some plugins from the Permit extension point
+  may send a request for an operation requiring to wait for a condition
+  (e.g. wait for additional resources to be available or wait for all pods
+  in a gang to be assumed).
+  The `WaitOnPermit` extension point allows you to wait for such a condition to be met.
+- [PreBind](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L580)
+- [Bind](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L592)
+- [PostBind](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L611)
+
+In case of processing of any of the extension points fails, `Unreserve` operation
+of all `Reserve` plugins is invoked (e.g. free resources allocated for a gang of pods).
+
+## Configuring and assembling the scheduler
+
+The scheduler codebase spans across various locations. Last but not least to mention:
+- [cmd/kube-scheduler/app](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/cmd/kube-scheduler/app):
+  location of the controller code alongside definition of CLI arguments (honors the standard setup for all Kubernetes controllers)
+- [pkg/scheduler](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler):
+  the default scheduler codebase root directory
+- [pkg/scheduler/core](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/core):
+  location of the default scheduling algorithm
+- [pkg/scheduler/framework](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/framework):
+  scheduling framework alongside plugins
+- [pkg/scheduler/internal](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/internal):
+  implementation of the cache, queues and other internal elements
+- [staging/src/k8s.io/kube-scheduler](https://github.com/kubernetes/kubernetes/tree/a651804427dd9a15bb91e1c4fb7a79994e4817a2/staging/src/k8s.io/kube-scheduler):
+  location of ComponentConfig API types
+
```

Comment:
Also mention the test codebase? (ut, integration, benchmark, e2e)

Reply:
Which unit tests do you have in mind?

Reply:
I mean the regular ones colocated with the source code (*_test.go)

Reply:
Are those worth mentioning given they are already part of the mentioned codebase locations? I can add another bullet saying "Regular unit tests ending with `_test.go` located alongside mention codebase locations". Though given unit tests are more or less must for every "serious" implementation, I consider them already included.